tmt/utils/templates.py

```diff
@@ -0,0 +1,362 @@
+"""
+Template rendering.
+
+Package provides primitives for Jinja2 template rendering, plus our
+custom filters.
+"""
+
+import re
+import textwrap
+from re import Match
+from typing import (
+    Any,
+    Callable,
+    Optional,
+    cast,
+    )
+
+import fmf
+import fmf.utils
+import jinja2
+import jinja2.exceptions
+
+from tmt.utils import GeneralError, Path
+
+
+def _template_filter_basename(  # type: ignore[reportUnusedFunction,unused-ignore]
+        pathlike: str) -> str:
+    """
+    Return the last component of the given path.
+
+    .. code-block:: jinja
+
+        # "/etc/fstab" -> "fstab"
+        {{ "/etc/fstab" | basename }}
+
+        # "/var/log/" -> "log"
+        {{ "/var/log/" | basename }}
+    """
+
+    return Path(pathlike).name
+
+
+def _template_filter_match(  # type: ignore[reportUnusedFunction,unused-ignore]
+        s: str, pattern: str) -> Optional[Match[str]]:
+    """
+    Return `re.Match`__ if the string matches a given pattern.
+
+    Pattern is tested in "match" mode, i.e. it must match from the
+    beginning of the string. See :ref:`regular-expressions` description
+    for more details.
+
+    __ https://docs.python.org/3.9/library/re.html#match-objects
+
+    .. code-block:: jinja
+
+         # 'foo/bar' -> 'foo/bar'
+        {{ 'foo/bar' | match('foo/.*').group() }}
+
+        # 'foo/bar' -> ''
+        {{ 'foo/bar' | regex_match('foo/(.+?)/(.*)') }}
+
+        # 'foo/bar/baz' -> 'bar'
+        {{ 'foo/bar' | regex_match('foo/(.+?)/.*').group(1) }}
+    """
+
+    return re.match(pattern, s)
+
+
+def _template_filter_search(  # type: ignore[reportUnusedFunction,unused-ignore]
+        s: str, pattern: str) -> Optional[Match[str]]:
+    """
+    Return `re.Match`__ if the string matches a given pattern.
+
+    Pattern is tested in "search" mode, i.e. it can match anywhere
+    in the string. See :ref:`regular-expressions` description for more
+    details.
+
+    __ https://docs.python.org/3.9/library/re.html#match-objects
+
+    .. code-block:: jinja
+
+         # '  foo/bar' -> 'foo/bar'
+        {{ '  foo/bar' | match('foo/.*').group() }}
+
+        # '  foo/bar' -> ''
+        {{ '  foo/bar' | regex_match('foo/(.+?)/(.*)') }}
+
+        # '  foo/bar/baz' -> 'bar'
+        {{ '  foo/bar' | regex_match('foo/(.+?)/.*').group(1) }}
```

Comment:
```suggestion
         # 'foo/bar' -> 'foo/bar'
        {{ 'foo/bar' | search('foo/.*').group() }}

        # 'foo/bar' -> ''
        {{ 'foo/bar' | regex_search('foo/(.+?)/(.*)') }}

        # 'foo/bar/baz' -> 'bar'
        {{ 'foo/bar' | regex_search('foo/(.+?)/.*').group(1) }}
```
At least I assume this meant to regex_search, not regex_match here.  
Indentation is inconsistent with the above and below docstrings.

Reply:
Nice, it was even messier. Hopefully addressed in https://github.com/teemtee/tmt/pull/3011.

Reply:
@martinhoyer is it possible to resolve this conversation?

Reply:
Apologies!